app/models/concerns/imageable.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+module Imageable
+  extend ActiveSupport::Concern
+
+  included do
+    has_many :images, as: :imageable, dependent: :destroy
```

Comment:
Paperclip take care about actual files, so both actual files and database handles will be deleted. I doubt it myself, but it just works.